mrp_stock_actual_date/models/stock_move.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    @api.depends(
+        "date",
+        "picking_id.actual_date",
+        "scrap_ids.actual_date",
+        "raw_material_production_id.actual_date",
+        "production_id.actual_date",
+        "consume_unbuild_id.actual_date",
+        "unbuild_id.actual_date",
+    )
+    def _compute_actual_date(self):
```

Comment:
This compute method is getting too bloated.

Can we refactor the logic by adding a new field, say, `actual_date_source` to `stock.move`?

- `actual_date_source` should just replicate the `actual_date` value of the "parent" model (`stock.picking`, `mrp.production` or `mrp.unbuild`) by extending the `write()` method of each model
- `_compute_actual_date()` should be kept simple by using `actual_date_source` as a trigger, instead of all the `actual_date` fields of the "parent" models.